// @lnu/eslint-config/index.js
/**
 * This file defines the configuration options for ESLint.
 *
 * @author Johan Leitet <johan.leitet@lnu.se>
 * @author Mats Loock <mats.loock@lnu.se>
 * @version 1.1.4
 */

module.exports = {
  root: true,
  env: {
      "es2022": true,
      "browser": true,
      "node": true,
      "jest": true
  },
  extends: [
    'standard'
  ],
  plugins: [
    'jsdoc'
  ],
  rules: {
    'jsdoc/check-alignment': 1, // Recommended
    'jsdoc/check-examples': 0, // 1 overridden (This rule currently does not work in ESLint 8 (we are waiting for issue 14745).)
    'jsdoc/check-indentation': 1,
    'jsdoc/check-param-names': 2, // Overridden
    'jsdoc/check-syntax': 1,
    'jsdoc/check-tag-names': 1, // Recommended
    'jsdoc/check-types': 1, // Recommended
    'jsdoc/implements-on-classes': 1, // Recommended
    'jsdoc/match-description': 1,
    'jsdoc/newline-after-description': 1, // Recommended
    'jsdoc/no-types': 0, // Overridden
    'jsdoc/no-undefined-types': 1, // Recommended
    'jsdoc/require-description': 2,
    'jsdoc/require-description-complete-sentence': 0, // 1 overridden
    'jsdoc/require-example': 0, // Overridden
    'jsdoc/require-hyphen-before-param-description': 0, // Overridden
    'jsdoc/require-jsdoc': [
      2, // Overridden the Recommended
      {
        require: {
          ArrowFunctionExpression: true, // Overridden the Recommended
          ClassDeclaration: true, // Overridden the Recommended
          ClassExpression: true, // Overridden the Recommended
          FunctionDeclaration: true,
          FunctionExpression: true, // Overridden the Recommended
          MethodDefinition: true// Overridden the Recommended
        }
      }
    ],
    'jsdoc/require-param': 2, // Overridden the Recommended
    'jsdoc/require-param-description': 1, // Recommended
    'jsdoc/require-param-name': 2, // Overridden the Recommended
    'jsdoc/require-param-type': 2, // Overridden the Recommended
    'jsdoc/require-returns': 2, // Overridden the Recommended
    'jsdoc/require-returns-check': 2, // Overridden the Recommended
    'jsdoc/require-returns-description': 1, // Recommended
    'jsdoc/require-returns-type': 2, // Overridden the Recommended
    'jsdoc/valid-types': 1 // Recommended
  }
}
